external/fv3fit/fv3fit/keras/_models/shared/loss.py

```diff
@@ -0,0 +1,86 @@
+import dataclasses
+import numpy as np
+import tensorflow as tf
+
+
+def _standard_scaled_mse(std):
+    std = tf.constant(std, dtype=std.dtype)
+
+    def custom_loss(y_true, y_pred):
+        return tf.math.reduce_sum(
+            tf.math.reduce_mean(tf.math.square((y_pred - y_true) / std), axis=0)
+        )
+
+    return custom_loss
+
+
+def _standard_scaled_mae(std):
+    std = tf.constant(std, dtype=std.dtype)
+
+    def custom_loss(y_true, y_pred):
+        return tf.math.reduce_sum(
+            tf.math.reduce_mean(tf.math.abs((y_pred - y_true) / std), axis=0)
+        )
+
+    return custom_loss
+
+
+def multiply_loss_by_factor(original_loss, factor):
+    def loss(y_true, y_pred):
+        return tf.math.scalar_mul(factor, original_loss(y_true, y_pred))
+
+    return loss
+
+
+def _uniform_scaled_mse(std):
+    factor = tf.constant(1.0 / np.mean(std ** 2), dtype=std.dtype)
+    return multiply_loss_by_factor(tf.losses.mse, factor)
+
+
+def _uniform_scaled_mae(std):
+    factor = tf.constant(1.0 / np.mean(std), dtype=std.dtype)
+    return multiply_loss_by_factor(tf.losses.mae, factor)
+
+
+@dataclasses.dataclass
+class LossConfig:
+    """
+    Attributes:
+        loss_type: one of "mse" or "mae"
+        scaling: "standard" corresponds to scaling each feature's lossby
+            its scale, "standard_uniform" corresponds to scaling
+            each feature's loss by the mean of all feature scales, where
+            the scale is variance for MSE loss or standard deviation
+            for MAE loss
+        weight: A scaling factor by which to modify this loss
+    """
```

Comment:
The `DenseModel` has a `normalize_loss` option that defaults to True- I don't know if we've ever used the `False` option, but you may want to keep it in the `LossConfig` and add a test.

Reply:
I decided not to implement "no normalization" because we've never had a use case for it or even discussed possibly using it. When I added it to DenseModel I probably shouldn't have because of YAGNI.